### ISSUE TITLE
Should there be a set of preconfigured providers based on <$env:keys>

### ISSUE THREAD

**Issue by dfinke** [dfinke/PSAI#54]

**Comment by Agazoth** [comment]
A good question.

Do we want to hard-code a specific set of providers in the module?

We would have to define a list of supported env:keys that we then include. That can be done with some sort of pre-fixing $env:PSAI-OpenAI-APIKey. Other required settings like baseUri, ModelName and Version would need similar env:keys if we want to avoid having to add additional code to Set-OAIProvider and Set-AzOAISecrets. Then functions could read all the $env:PSAI variables and trigger different behavior on that. This would not solve the issue of variations in the message and answer structures.

Im my opinion keys and settings should be handled in the Plugins (the AIProviders and their models) along with the individual peculiarites of a given Provider/Plugin. OpenAI and AzureOpenAI are very similar, but even the two have differences. OpenAI requires the model name in the body when posting to the backend, AzureOpenAI has the model in the uri. Anthropic is a completely different beast, that requires a completely different body on post and has different roles.

**Comment by Agazoth** [comment]
Related to #53

**Comment by dfinke** [comment]
Not for Azure. For some of the common providers and models, so consumers need only provide their key, preferrably via `$env`.

Aligns with reduced friction. Easy to get started, simple to expand for more advanced scenarios.

**Comment by Agazoth** [comment]
Should there be some sort of prefix like $env:PSAI-<APIKey> or should all included providers have a hard coded $env:<Provider>APIKey vairable?  The more generic approach would allow AIProvider creators to test the functionality

**Comment by dfinke** [comment]
I would follow the "docs" for each player - OpenAI, Anthropic, Google, etc. Use their naming, folks may come from different SDKs to this.

OpenAI needs to be updated to have two. I named it `$env:OpenAIKey` - I created that so it was what I expected in PS. It should also support OpenAI's docs.

**Comment by Agazoth** [comment]
This looks really familiar: https://llm.datasette.io/en/latest/setup.html#api-keys

I guess the env keys are also saved in the keys.json file. 

There must be other config info there. How else could the model know where to go with the "personal" key. 

Installing a provider must bring along the name for the env key. Combined with the keys.json that could be brought to autoload when the module is imported.

**Comment by dfinke** [comment]
Agreed! Going deeper on that code will have clues and may be a good architectural approach.
Or show the potential dead ends.
